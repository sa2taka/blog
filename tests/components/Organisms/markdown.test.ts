import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import Markdown from '../../../components/Organisms/markdown.vue';
import { posts } from '../../__mocks__/postData';

const localVue = createLocalVue();

jest.mock('contentful', () => ({
  createClient: (_: any) => {},
}));

describe('components/organisms/markdown', () => {
  let vuetify: any;
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const mountWithVuetify = (options?: any) => {
    return mount(Markdown, {
      localVue,
      vuetify,
      ...options,
    });
  };

  test('should render', () => {
    const markdown = posts[0].fields.body;
    const wrapper = mountWithVuetify({
      propsData: {
        markdown,
      },
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
