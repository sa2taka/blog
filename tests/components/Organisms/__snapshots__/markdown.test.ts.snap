// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/index should render 1`] = `
<div>
  <h1 id="テストです">テストです</h1>
  <h2 id="テストです">テストです</h2>
  <p>引用符</p>
  <blockquote>
    <ul>
      <li>list1</li>
    </ul>
  </blockquote>
  <ul>
    <li>list2<ul>
        <li>nested 2-1</li>
        <li>nested 2-2<ul>
            <li>nested 2-2-1</li>
            <li>nested 2-2-2</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>list3</li>
  </ul>
  <ol>
    <li>numberlist1</li>
    <li>numberlist2<ol>
        <li>nested number 2-1</li>
        <li>nested number 2-2<ol>
            <li>nested number 2-2-1</li>
            <li>nestednumber 2-2-2</li>
          </ol>
        </li>
      </ol>
    </li>
    <li>numberlist3</li>
  </ol>
  <p><a href="https://sa2taka.com" title="りんく">リンクテキスト</a></p>
  <p><del>打消し戦</del></p>
  <p><code class="hljsspan">code</code></p>
  <div class="filename">hoge.rb</div><code class="hljs ruby padding-for-filename">import <span class="hljs-string">'foo'</span>

puts <span class="hljs-string">'hoge'</span>

<span class="hljs-function"><span class="hljs-keyword">def</span> <span class="hljs-title">Fuga</span></span>
  foobar()
<span class="hljs-keyword">end</span></code>
  <hr>
  <table>
    <thead>
      <tr>
        <th>Header</th>
        <th>Header</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Cell</td>
        <td>Cell</td>
      </tr>
      <tr>
        <td>Cell</td>
        <td>Cell</td>
      </tr>
    </tbody>
  </table>
</div>
`;
